FrontEnd/src/Components/Profile/StudentSection.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import AddIcon from '@mui/icons-material/Add';
+import StudentItem from './StudentItem';
+import StudentProfile from '../../Pages/ProfilePage/StudentProfile';
+import EditStudentProfile from '../../Pages/ProfilePage/EditStudentProfile';
+import DeleteDialog from './DeleteDialog';
+import { deleteStudent, findStudents } from '../../api/students';
+import { defaultStudentInfo } from '../../utils/constants';
+
+function StudentSection({
+  user, students, setStudents, setSuccessOpen, setErrorOpen, setInfoOpen, setAlertMessage,
+}) {
+  const [openEditModal, setOpenEditModal] = useState(false);
+  const [currentStudent, setCurrentStudent] = useState(defaultStudentInfo);
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [addStudent, setAddStudent] = useState(false);
+
+  const handleOpenDialog = () => {
+    setOpenDeleteDialog(true);
+  };
+
+  const handleCloseDialog = () => {
+    setOpenDeleteDialog(false);
+  };
+
+  const handleEditStudent = (student) => {
+    setCurrentStudent(student);
+    setOpenEditModal(!openEditModal);
+  };
+
+  const handleDeleteCurrentStudent = () => {
+    handleCloseDialog();
+    deleteStudent({ _id: currentStudent._id })
+      .then((data) => {
+        console.log(data);
+        if (data.status === 400) {
+          setAlertMessage('Se produjo un error al eliminar al estudiante.');
+          setErrorOpen(true);
+        } else {
+          setAlertMessage('Estudiante eliminado correctamente.');
+          setSuccessOpen(true);
+        }
+      })
+      .finally(() => {
+        findStudents({ idUser: user.sub })
+          .then((response) => response.json())
+          .then((data) => {
+            console.log(data);
+            setStudents(data);
+          });
+      });
+  };
+
+  return (
+    <>
+      <Box
+        sx={{
+          pt: 2,
+          display: user?.p99roles.length === 0 ? 'flex' : 'none',
```

Comment:
```
user?.p99roles.length === 0 
```
I see this check multiple times in this document, you should be able to use this only once or at least make it so its not checked that often

Reply:
I switched it to a constant